lib/Parser/Parse.cpp

```diff
@@ -5265,10 +5265,11 @@ bool Parser::ParseFncDeclHelper(ParseNodePtr pnodeFnc, LPCOLESTR pNameHint, usho
             m_currentNodeNonLambdaDeferredFunc = pnodeFncSaveNonLambda;
         }
         m_currentNodeDeferredFunc = pnodeFncSave;
-        if (m_currentNodeFunc && pnodeFnc->sxFnc.HasWithStmt())
-        {
-            GetCurrentFunctionNode()->sxFnc.SetHasWithStmt(true);
-        }
+    }
```

Comment:
Yeah, I just was about to say :)

The other thing I noted was that we propagated other information similarly but in the caller of `ParseFncDeclHelper`, instead of here.  Would it make sense to consolidate all these flag propagations together in the same place?